networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/Eth2OutgoingRequestHandler.java

```diff
@@ -76,37 +76,29 @@ public void handleInitialPayloadSent(final RpcStream stream) {
   }
 
   @Override
-  public void processInput(
-      final NodeId nodeId, final RpcStream rpcStream, final InputStream input) {
+  public void active(NodeId nodeId, RpcStream rpcStream) {}
+
+  @Override
+  public void processData(final NodeId nodeId, final RpcStream rpcStream, final ByteBuf data) {
     try {
       this.rpcStream = rpcStream;
 
-      Optional<TResponse> maybeResponse =
-          responseDecoder.decodeNextResponse(input, this::onFirstByteReceived);
-      while (!isClosed.get() && maybeResponse.isPresent()) {
-        final TResponse response = maybeResponse.get();
-        responseProcessor.processResponse(response);
-
-        final int chunksReceived = currentChunkCount.incrementAndGet();
-        if (chunksReceived >= maximumResponseChunks) {
-          // Make sure there aren't any trailing unconsumed bytes
-          if (input.available() > 0) {
-            LOG.debug(
-                "Encountered unconsumed data after last expected response chunk was processed.");
-            cancelRequest(rpcStream, RpcException.EXTRA_DATA_APPENDED);
-          } else {
-            completeRequest(rpcStream);
-          }
-          break;
+      if (!isClosed.get()) {
+        List<TResponse> maybeResponses =
+            responseDecoder.decodeNextResponses(data, this::onFirstByteReceived);
```

Comment:
I think we can simplify this by handling the "first byte" logic here:
```suggestion
        if (data.isReadable()) {
            onFirstByteReceived();
        }
        List<TResponse> maybeResponses = responseDecoder.decodeNextResponses(data);
```

Reply:
Yep, good suggestion 👍